pkgs/development/tools/parsing/tree-sitter/grammar.nix

```diff
@@ -6,19 +6,14 @@
 
 # Build a parser grammar and put the resulting shared object in `$out/parser`
 
-{
-  # language name
-  language
-, version
+{ version
 , src
 , location ? null
 , generate ? false
 , ...
 }@args:
 
 stdenv.mkDerivation ({
-  pname = "${language}-grammar";
-
   inherit src version;
```

Comment:
The `buildGrammar` function: https://github.com/NixOS/nixpkgs/blob/932390c76b7e0d44656a648fe336a706f659cd42/pkgs/development/tools/parsing/tree-sitter/default.nix#L52

is not only used in `pkgs/development/tools/parsing/tree-sitter/default.nix` but also used elsewhere, for example, `neovim-unwrapped`: https://github.com/NixOS/nixpkgs/blob/daeb19c6a7edabb2a7417dcf0bc75095b7403b70/pkgs/by-name/ne/neovim-unwrapped/package.nix#L175

These external usages may pass `language` as an argument.


That means, since we removed `language` from the function argument, and removed `pname = "${language}-grammar";`, there's no way for external usages of `buildGrammar` to generate a `pname` for new grammars, causing `name` not being generated, further causing eval errors